core/data-plane/data-plane-core/src/main/java/org/eclipse/edc/connector/dataplane/framework/DataPlaneDefaultIamServicesExtension.java

```diff
@@ -62,7 +63,13 @@ public DataPlaneAccessControlService defaultAccessControlService(ServiceExtensio
 
     @Provider(isDefault = true)
     public DataPlaneAccessTokenService defaultAccessTokenService(ServiceExtensionContext context) {
-        return new DefaultDataPlaneAccessTokenServiceImpl(new JwtGenerationService(), accessTokenDataStore, context.getMonitor().withPrefix("DataPlane IAM"), getPrivateKeySupplier(context), tokenValidationService, localPublicKeyService);
+        return new DefaultDataPlaneAccessTokenServiceImpl(new JwtGenerationService(),
+                accessTokenDataStore, context.getMonitor().withPrefix("DataPlane IAM"),
+                getPrivateKeySupplier(context), publicKeyIdSupplier(context), tokenValidationService, localPublicKeyService);
+    }
+
+    private Supplier<String> publicKeyIdSupplier(ServiceExtensionContext context) {
+        return () -> context.getConfig().getString(TOKEN_VERIFIER_PUBLIC_KEY_ALIAS);
```

Comment:
maybe the key could be extracted and verified in the `@Provider` method, so it will avoid lazy runtime errors

Reply:
I would do this in another PR, same as the private key alias once we have the dataplane signaling as default. I will raise an issue about this